kinder/ci/tools/update-workflows/templates/workflows/instance-config-tasks.yaml

```diff
@@ -0,0 +1,322 @@
+  - --control-plane-nodes={{ .vars.controlPlaneNodes }}
+  - --worker-nodes={{ .vars.workerNodes }}
+  - --loglevel=debug
+  timeout: 5m
+- name: init
+  description: |
+    Initializes the Kubernetes cluster with version "initVersion"
+    by starting the boostrap control-plane nodes
+  cmd: kinder
+  args:
+  - do
+  - kubeadm-init
+  - --name={{ .vars.clusterName }}
+  - --loglevel=debug
+  - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+  - --kubeadm-feature-gate="NodeLocalCRISocket=true"
+  - --copy-certs=auto
+  timeout: 5m
+- name: join
+  description: |
+    Join the other nodes to the Kubernetes cluster
+  cmd: kinder
+  args:
+  - do
+  - kubeadm-join
+  - --name={{ .vars.clusterName }}
+  - --loglevel=debug
+  - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+  - --kubeadm-feature-gate="NodeLocalCRISocket=true"
+  - --copy-certs=auto
+  timeout: 10m
+- name: check-annotation
+  description: |
+    Check that the cri-socket annotation does not exist on each node.
+  cmd: /bin/bash
+  args:
+    - -c
+    - |
+      set -x
+
+      docker exec {{ .vars.clusterName }}-control-plane-1 bash -c '
+        nodes=$(KUBECONFIG=/etc/kubernetes/admin.conf kubectl get nodes -o jsonpath='{.items[*].metadata.name}')
+
+        for node in $nodes; do
+          annotation=$(KUBECONFIG=/etc/kubernetes/admin.conf kubectl get node "$node" -o jsonpath="{.metadata.annotations['kubeadm\.alpha\.kubernetes\.io/cri-socket']}")
+
+          if [ -z "$annotation" ]; then
+            echo "Annotation not found on node $node"
+          else
+            echo "Annotation found on node $node: $annotation"
+            exit 1
+          fi
+        done
+      ' || {
+        echo "Command failed for $node. Exiting.";
+        exit 1;
+        }
+
+      exit 0
+  timeout: 5m
+- name: check-config
+  description: |
+    Check if the kubelet instance configuration exists on each node.
+  cmd: /bin/bash
+  args:
+    - -c
+    - |
+      set -x
+
+      for node in control-plane-{1..3} worker-{1..2}; do
+        CMD="docker exec {{ .vars.clusterName }}-$node"
+
+        ${CMD} bash -c '
+          if [[ -f "/var/lib/kubelet/instance-config.yaml" ]] && grep -q "containerRuntimeEndpoint:" /var/lib/kubelet/instance-config.yaml; then
+            echo "File exists and contains 'containerRuntimeEndpoint:' field."
+          else
+            echo "File does not exist or does not contain 'containerRuntimeEndpoint:' field."
+            exit 1
+          fi
+        ' || {
+        echo "Command failed for $node. Exiting.";
+        exit 1;
+        }
+      done
+
+      exit 0
+  timeout: 5m
+- name: upgrade
+  description: |
+    upgrades the cluster to Kubernetes "upgradeVersion"
+  cmd: kinder
+  args:
+  - do
+  - kubeadm-upgrade
+  - --upgrade-version={{ .vars.upgradeVersion }}
+  - --name={{ .vars.clusterName }}
+  - --loglevel=debug
+  - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+  - --kubeadm-feature-gate="NodeLocalCRISocket=true"
```

Comment:
also not needed. `kubeadm upgrade` downloads the kubeadm-config config map and construct a ClusterConfiguration from it. then all places in upgrade would know about the current list of enabled feature gates.